src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py

```diff
@@ -0,0 +1,69 @@
+"""
+Thread used to monitor the agent draining process. Should
+eventually report no issues meaning that the agent is ready
+to be shutdown and a new version be put in place.
+"""
+from __future__ import division
+
+__all__ = []
+
+import logging
+import threading
+from WMCore.WorkerThreads.BaseWorkerThread import BaseWorkerThread
+from WMComponent.AnalyticsDataCollector.DataCollectAPI import isDrainMode
+from WMComponent.DBS3Buffer.DBSBufferUtil import DBSBufferUtil
+
+class DrainStatusPoller(BaseWorkerThread):
+    """
+    Collects information related to the agent drain status
+    """
+    # class variable that contains drain statistics
+    drainStats = {}
```

Comment:
AgentStatusWatcher spawns three threads based of the AgentStatusPoller, DrainStatusPoller and ResourceControlUpdater classes.  Each instance of those classes doesn't know about the other.  AgentStatusPoller needs to get variable contents from DrainStatusPoller, hence the class variable.  If there is a better way, I'm open to suggestions.